File: CSharp/01. Two Sum (Easy)/Solution2.cs
using System;
using System.Collections.Generic;

namespace LeetCode
{
	public class Solution2
	{
		public int[] TwoSum(int[] nums, int target)
		{
			if (nums == null)
				return null;

			var index = new Dictionary<int, int>();

			for (int i = 0; i < nums.Length; i++) {
				int ni = nums[i];
				int n1 = target - ni;
				if (index.ContainsKey(n1)) {
					return new int[] {index[n1] , i };
				}
				if (!index.ContainsKey(ni)) {
					index.Add(ni, i);
				}
			}

			return null;
		}
	}
}

File: Java/01. Two Sum (Easy)/src/LeetCode/TwoSum/Solution2.java
package LeetCode.TwoSum;
import java.util.HashMap;


public class Solution2
{
	public int[] twoSum(int[] nums, int target)
	{
		if (nums == null)
			return null;

		HashMap<Integer, Integer> index = new HashMap<Integer, Integer>();

		for (int i = 0; i < nums.length; i++) {
			int ni = nums[i];
			int n1 = target - ni;
			if (index.containsKey(n1)) {
				return new int[] {index.get(n1) , i};
			}
			if (!index.containsKey(ni)) {
				index.put(ni, i);
			}
		}

		return new int[0];
	}
}
